lgc/patch/FragColorExport.cpp

```diff
@@ -502,7 +502,17 @@ bool LowerFragColorExport::runOnModule(Module &module) {
   if (!fragEntryPoint)
     return false;
 
-  auto retInst = fragEntryPoint->back().getTerminator();
+  auto &bbList = fragEntryPoint->getBasicBlockList();
+  ReturnInst *retInst = nullptr;
+  for (auto bb = bbList.rbegin(); bb != bbList.rend(); ++bb) {
+    auto *terminator = bb->getTerminator();
+    if (isa<ReturnInst>(terminator)) {
+      retInst = dyn_cast<ReturnInst>(terminator);
+      break;
+    }
+  }
```

Comment:
I think more idiomatic code would look something like this:

```
ReturnInst *retInst = nullptr;
for (auto &block : llvm::reverse(*fragEntryPoint)) {
  if (auto ret = dyn_cast<ReturnInst>(block.getTerminator())) {
    retInst = ret;
    break;
  }
}
```

Reply:
Done

Reply:
Oh, I missed that part.  That is nice.

Reply:
Thanks